### ISSUE TITLE
Improved chat UI in mobile environment

### ISSUE THREAD

**Issue by leafeon012** [owncast/owncast#3931]
### Share your bug report, feature request, or comment.

![Screenshot_20240923_002922_Iceraven](https://github.com/user-attachments/assets/a95ced40-b55d-467d-9177-f40d31e3ed60)

When open a chat on mobile, the screen background darkens and the video is obscured.

Can watch a video while chatting at the same time? 

Are there any suitable ways to improve this?

**Comment by gabek** [comment]
This was the solution for mobile out of desperation, not because it's optimal. There are no plans to redesign the frontend anytime soon, as that already happened. But I'd 100% be up for it if somebody wanted to help work on mobile frontend UI.

**Comment by germainelee** [comment]
I did a mockup in penpot of how it would look like potentially.

![Image](https://github.com/user-attachments/assets/7d3aec05-3a40-4c56-be3c-09441bbaae1c)

**Comment by gabek** [comment]
This would be slightly better, but only a very little. The real solution is how it used to be, that on mobile, chat is a tab that a user can select, and have it take up the bottom part of the screen where the info or followers are displayed. However, a number of people worked on this, and the layout issues were too numerous. 

If we had unlimited time and somebody who really loved focusing on frontend who wanted to tackle this, we could try this again. 

For reference

- Chat used to be a mobile tab see https://github.com/owncast/owncast/issues/2991#issue-1698749204 as an example of where it was.
- Some of the issues that it caused can be seen here https://github.com/owncast/owncast/pull/3022#issuecomment-1552157497, as different parts of the page needed to scroll independently, due to the chat needing to scroll independently. 
- The PR that changed mobile chat into a modal https://github.com/owncast/owncast/pull/3038 because I had given up.

<img width="40%" alt="Image" src="https://github.com/user-attachments/assets/11fb5b93-8254-4cea-bb88-3700225077ed" />

**Comment by germainelee** [comment]
The current chat modal implementation is fine imo, as it looks like it can take into account the keyboard space with some tweaks. Removing the overlay would allow viewers to continue watching the stream clearly

Is it possible to remove the overlay when the chat expands?

More importantly, can the chat height be "dynamic" Such that it wont cover the entire screen when the keyboard expands?
![Screenshot_20241204_190917_Chrome.jpg](https://github.com/user-attachments/assets/c6e0ed73-91dc-4dad-91e8-05e8e91ae6ac)

**Comment by gabek** [comment]
> Is it possible to remove the overlay when the chat expands?

That should be doable, even if we have to force it via CSS.

> More importantly, can the chat height be "dynamic" Such that it wont cover the entire screen when the keyboard expands?

This would be more of a challenge. If you wanted the keyboard to be displayed, and the text input to be displayed, and the video to be displayed, that's pretty much the entire screen (on some devices) with almost no room for the actual chat.

![Image](https://github.com/user-attachments/assets/b37851a7-d984-4ac0-baec-a7673cfb45a6)

**Comment by germainelee** [comment]
> that's pretty much the entire screen (on some devices) with almost no room for the actual chat.

yes, especially on smaller devices. Do you think Picture-in-picture (PiP) mode makes sense, with the chat behind and video in front?
Or shrink the video height?

**Comment by gabek** [comment]
So I'm all for brainstorming, but as far as I know, not many people use the Owncast interface on mobile. And it's many times more of the effort. So any major changes are unlikely to get done by me. I really dislike doing layout work, and mobile web layout work is the worst version of that. You can read some of the old threads dealing with mobile layout and see my frustration and how bad things were going. I really don't want to go back there.

**Comment by germainelee** [comment]
sure I understand where you are coming from. 

> with almost no room for the actual chat.

Ah right. Scratch the dynamic height then. Just remove the overlay, like the mockup in https://github.com/owncast/owncast/issues/3931#issuecomment-2516063829, so that viewer can watch the stream and read chat at the same time.

Due to limited screen estate on mobile, it'd impossible to watch video, type in chat with keyboard expanded on a web browser in mobile.
<sub><sup>anything that has to do w double scrolling is a b*tch and i also dun recommend it.</sup></sub>

**Comment by germainelee** [comment]
> not many people use the Owncast interface on mobile

Is there analytics captured for this?

**Comment by gabek** [comment]
We can't capture analytics. All we have is worth of mouth.

**Comment by germainelee** [comment]
3 issues raised by 3 different people, including this issue, requesting that the chat to not overlap the video as viewers would like to view the stream and the chat at the same time.

**Comment by gabek** [comment]
I totally get it. There's just no plan to rebuild the mobile layout. It was a huge amount of work to get it where it is, and this is as good as it could get. This isn't what I'm going to focus on in the near term. I don't enjoy that work, and after v0.1.0 it's not going to be rebuilt again. I can't overstate how much work and how long the current layout took, and that was with multiple people helping out. Right now it's just me.

Plus like above, if we showed full video and full chat it would look like this:

<img width="40%" alt="Image" src="https://github.com/user-attachments/assets/bacfece0-b045-4651-9d83-43c051a3edfe" />

That's like one line of chat. There's just not enough room for what's being asked.

**Comment by gabek** [comment]
For reference, this is how Twitch does chat on mobile web. It hides video so chat fits.

<img src="https://github.com/user-attachments/assets/1364c518-f748-4748-9e23-0c061cb295b0" width="40%">

**Comment by germainelee** [comment]
the chat can cover the video when keyboard is expanded. From what I understand, they are requesting to put the video and chat side by side (keyboard collapsed).
